client/src/components/Workflow/Editor/Forms/FormOutputLabel.vue

```diff
@@ -44,7 +44,22 @@ const label = computed(() => {
     return null;
 });
 
-function onInput(newLabel: string) {
+function onInput(newLabel: string | undefined | null) {
+    console.log("onInput Fires", newLabel);
+    if (newLabel === "") {
+        // user deleted existing label, we inactivate this as output
+        const strippedWorkflowOutputs = (props.step.workflow_outputs || []).filter(
+            (workflowOutput) => workflowOutput.output_name !== props.name
+        );
+        stepStore.updateStep({ ...props.step, workflow_outputs: strippedWorkflowOutputs });
+        error.value = undefined;
```

Comment:
Maybe move `error.value = undefined;` at the top and remove the others?

Reply:
That's a good catch, I think the null/undefined case shouldn't reset the state. I've kept the two separate calls, seems marginally easier to grasp than setting error to undefined and then setting it again if the output is a duplicate.